modules/baremetal/main.tf

```diff
@@ -2,12 +2,35 @@ data "ibm_is_subnet" "subnet" {
   identifier = var.subnet_id
 }
 
+# Primary reserved IP
+resource "ibm_is_subnet_reserved_ip" "bms_primary_reserverd_ip" {
+  count       = var.manage_reserved_ips ? 1 : 0
+  name        = "${var.name}-prim-reserve-ip"
+  subnet      = var.subnet_id
+  auto_delete = false
+}
+
+# Secondary reserved IP
+resource "ibm_is_subnet_reserved_ip" "bms_secondary_reserverd_ip" {
+  count       = var.manage_reserved_ips && var.secondary_vni_enabled ? 1 : 0
+  name        = "${var.name}-sec-reserve-ip-ip"
```

Comment:
I think you included "-ip" twice here by accident.

Also double check if there is a character limit for the name of a reserved IP. If it is a shorter limit, you may have to restrict length of var.name, or shorten this suffix, or both, so that you do not exceed the limit.